ibm/service/contextbasedrestrictions/resource_ibm_cbr_zone.go

```diff
@@ -291,24 +305,58 @@ func resourceIBMCbrZoneCreate(context context.Context, d *schema.ResourceData, m
 	return resourceIBMCbrZoneRead(context, d, meta)
 }
 
+// waitForCbrZoneRead will leverage use retry.StateChangeConf due to the service's eventual consistency
+func waitForCbrZoneRead(cbrClient *contextbasedrestrictionsv1.ContextBasedRestrictionsV1, context context.Context, id string, timeout time.Duration) (interface{}, error) {
+	stateConf := &retry.StateChangeConf{
+		Pending: []string{cbrRuleReadPending},
+		Target:  []string{cbrRuleReadError, cbrRuleReadComplete, ""},
+		Refresh: func() (interface{}, string, error) {
+			log.Printf("[INFO] Retrying cbr rule (%s) read", id)
+			getZoneOptions := &contextbasedrestrictionsv1.GetZoneOptions{}
+			getZoneOptions.SetZoneID(id)
+			_, response, err := cbrClient.GetZoneWithContext(context, getZoneOptions)
+			if err != nil && response.StatusCode == 404 {
```

Comment:
better to have log output with error and waiting time for next try if the `retry` does not do it